src/shared/utils/helpers/get-query-params.ts

```diff
@@ -1,21 +1,28 @@
-import { isBrowser } from "@utils/browser";
+type Empty = NonNullable<unknown>;
+
+type QueryMapping<PropsT, FallbacksT extends Empty> = {
+  [K in keyof PropsT]-?: (
+    input: string | undefined,
+    fallback: K extends keyof FallbacksT ? FallbacksT[K] : undefined,
+  ) => PropsT[K];
+};
 
 export default function getQueryParams<
-  T extends Record<string, any>,
->(processors: {
-  [K in keyof T]: (param: string) => T[K];
-}): T {
-  if (isBrowser()) {
-    const searchParams = new URLSearchParams(window.location.search);
+  PropsT,
+  FallbacksT extends Empty = Empty,
+>(
+  processors: QueryMapping<PropsT, FallbacksT>,
+  source: string | undefined,
+  fallbacks: FallbacksT,
```

Comment:
A default value would be good here. I noticed that you pass an empty object as the third arg to this function practically everywhere you call it.
```suggestion
  fallbacks: FallbacksT = {},
```